app/controllers/pokemon_controller.rb

```diff
@@ -15,97 +15,55 @@ class PokemonController < ApplicationController
-        if @pokemon.next_page
-          next_page_url = pokemon_url(page: @pokemon.next_page)
-          @res[:next_page] = next_page_url
-        end
-        @res.merge!(@pokeList)
-        render json: @res, status: :ok
+        render json: PokemonIndexService.res(@pokemon, Pokemon.count)
     end
   end
 
   # GET /pokemon/{id}
   def show
-    @poke = find_pokemon
-    @poke_json = @poke.slice(:id, :name, :type_1, :type_1, :type_2, :total, :hp, :attack,
-                :defense, :sp_atk, :sp_def, :speed, :generation, :legendary)
-            render json: @poke_json, status: :ok
+    @poke = PokemonFindService.find(Pokemon, params[:id])
+    if !@poke.nil?
+      @poke_json = @poke.slice(:id, :name, :type_1, :type_1, :type_2, :total, :hp, :attack,
+                  :defense, :sp_atk, :sp_def, :speed, :generation, :legendary)
+              render json: @poke_json
+    else
+      render json: { error: "Pokemon not found" }, status: :not_found
+    end
   end
 
   #  POST /pokemon
   def create
-    # puts "Create esta siendo llamado !!!"
     parsed_body = JSON.parse(request.body.read)
     # puts "Buscando ANDO params = #{parsed_body['pokemon']['name']}"
     poke_find =  Pokemon.find_by(name: parsed_body['pokemon']['name'])
   
     if poke_find.nil?
       # puts "NOOOOO ENcontrado"
-      @poke = Pokemon.create!(create_params)
+      @poke = Pokemon.create!(PokemonParams.create(params))
       @poke_json = @poke.slice(:id, :name, :type_1, :type_1, :type_2, :total, :hp, :attack,
           :defense, :sp_atk, :sp_def, :speed, :generation, :legendary)
       render json: @poke_json, status: :created
     elsif
-      # puts "ENcontradoNEW"
       render json: {error: "Pokemon exist"}, status: 409
     end
-    puts "despues IF"
-
   end
 
   # PUT /pokemon/{id} 
   def update
-    @poke = find_pokemon
-    @poke.update!(update_params)
-    @poke_json = @poke.slice(:id, :name, :type_1, :type_1, :type_2, :total, :hp, :attack,
-        :defense, :sp_atk, :sp_def, :speed, :generation, :legendary)
-    render json: @poke_json, status: :ok
-  end
-
-  def create_params
-    params.require(:pokemon).permit(:name, :type_1, :total, :hp, :attack,
-         :defense, :sp_atk, :sp_def, :speed, :generation, :legendary)
-  end
-
-  def update_params
-    params.require(:pokemon).permit(:name, :total, :hp, :attack, :defense, :sp_atk, :sp_def, :speed)
-  end
-
-  def find_pokemon
-    search_value = params[:id]
-
-    if search_value.to_i != 0
-        @poke = Pokemon.find_by(id: search_value)
-    else
-        # puts search_value.is_a?(String)
-        @poke = Pokemon.find_by(name: search_value)
-    end
-
-    if @poke == nil
-        render json: { error: "Pokemon not found" }, status: :not_found
-    else
-        return @poke  
+    @poke = PokemonFindService.find(Pokemon, params[:id])
+    if !@poke.nil?
+      @poke.update!(PokemonParams.update(params))
+      @poke_json = @poke.slice(:id, :name, :type_1, :type_1, :type_2, :total, :hp, :attack,
```

Comment:
Shoudve used the serializer instead

Reply:
Used -- Implemented with different attributes
Merci !!